### DISCUSSION TITLE
I want to distribute as an exe file, but I keep getting errors

### DISCUSSION THREAD

**Discussion by bizwsh2020** [dashingsoft/pyarmor#1156] [Q&A] [answered]
Yesterday, I purchased the paid version. When I use the command:

pyarmor gen -O obfdist --pack dist/foo foo.py

I encounter the following error:

CRITICAL unknown error, please check pyarmor.error.log
ERROR cannot import name 'PYZ_TYPE_PKG' from 'PyInstaller.loader.pyimod01_archive' (C:\python\lib\site-packages\PyInstaller\loader\pyimod01_archive.py)

When I follow the instructions from this site: https://pyarmor.readthedocs.io/en/latest/topic/repack.html
the build is successful, but the required libraries are not included in the build.

After obfuscating with PyArmor and then building with PyInstaller, the resulting .exe file produces the following error because essential libraries are not included:

Traceback (most recent call last):
  File "<frozen __main__>", line 3, in <module>
  File "<frozen Insert image into Excel>", line 3, in <module>
ModuleNotFoundError: No module named 'win32com'

pyarmor-7 pack works perfectly, but why was this feature removed?

Also, are you aware that if there is a comma (,) or space in the folder name, PyArmor's output goes to the wrong place? It seems like you've implemented a split function for commas and spaces.

**Comment by sSuHao** [comment, marked as answer]
I got the same error, did you find the solution?

Edit: Found the solution, upgrade to PyArmor 8.2.0

pip install -U pyarmor
